### PR TITLE
Review of 'Report Model Version and Configuration'

### PR DESCRIPTION
**Pull request by bateman**

### CODE REVIEW

File: guidelines/_sources/02_report-version-and-configuration.tex [hunk 1]
```diff
@@ -21,6 +21,28 @@ \subsubsection{Recommendations}
...
+``For fine-tuned models, researchers \textbf{must} additionally report:
+\begin{itemize}
+\item Base model used before fine-tuning
+\item Fine-tuning dataset characteristics (size, source, preprocessing steps)
+\item Fine-tuning hyperparameters (learning rate, epochs, batch size)
+\item Training objective (e.g., next token prediction, instruction tuning)
+\item Validation metrics used during fine-tuning
+\item If possible, a link to the fine-tuned model weights or a description of how to access them.''
+\end{itemize}
+}
```
Comment (sbaltes) [inline guidelines/_sources/02_report-version-and-configuration.tex:34]: This is a very good point. I'm now questioning our separation of model, config, architecture, and data.
@mircealungu What do you think? Does it make sense to merge the two sections? Fine tuning touches all of those aspects...

File: guidelines/_sources/02_report-version-and-configuration.tex [hunk 2]
```diff
@@ -37,6 +59,18 @@ \subsubsection{Example(s)}
 
 Our guidelines additionally suggest to report a checksum and exact dates, but otherwise this example is close to our recommendations. 
 
+\comment{Commercial tools have limited information. This is a bit tricky. What about a little more guidance?
+When studying commercial tools where configuration access is limited:
```
Comment (sbaltes) [inline guidelines/_sources/02_report-version-and-configuration.tex:63]: To be even more precise, one needs to distinguish between models (accessed via APIs) and tools. The list below contains both tools (Copilot) and commercial models (Codex).

File: guidelines/_sources/02_report-version-and-configuration.tex [hunk 3]
```diff
@@ -37,6 +59,18 @@ \subsubsection{Example(s)}
...
+\begin{itemize}
+\item Report product name, version number, and access date (e.g., "GitHub Copilot in Visual Studio Code v1.2.3, accessed between January-March 2024");
+\item Document any user-configurable settings and their values, such as temperature and maximum tokens (e.g., "For Codex, we set the temperature to 0.7, and the maximum number of tokens to 256.");
+\item Archive interaction logs showing tool behavior (see section \href{/guidelines/report-interaction-logs}{Report Interaction Logs});
+\item Acknowledge limitations in reproducibility due to the closed nature of the tool.
```
Comment (sbaltes) [inline guidelines/_sources/02_report-version-and-configuration.tex:68]: And I guess then also pointing to the corresponding section about limitations and mitigations?

File: guidelines/_sources/02_report-version-and-configuration.tex [hunk 4]
```diff
@@ -37,6 +59,18 @@ \subsubsection{Example(s)}
...
+\end{itemize}
+}
+
+\comment{Should we also address how API rate limits and usage constraints can affect experiments?}
```
Comment (sbaltes) [inline guidelines/_sources/02_report-version-and-configuration.tex:72]: That's a very good point, but maybe something rather for the limitations and mitigations section?

File: guidelines/_sources/02_report-version-and-configuration.tex [hunk 5]
```diff
@@ -60,6 +94,8 @@ \subsubsection{Study Types}
 The model name and date \must always be reported.
 In those cases, reporting other aspects such as prompts (\href{/guidelines/#report-prompts-and-their-development}{Report Prompts and their Development}) and interaction logs (\href{/guidelines//#report-interaction-logs}{Report Interaction Logs}) is essential.
 
+\comment{Beside guidelines I believe research would benefit from a structured format that tehy could easily follow, otherwise we risk inconsistent reporting across studies. Let's think about a template.}
```
Comment (sbaltes) [inline guidelines/_sources/02_report-version-and-configuration.tex:97]: Maybe we could develop such a template as part of https://github.com/se-ubt/llm-guidelines-internal/issues/9?

File: guidelines/_sources/02_report-version-and-configuration.tex (hunk 4 above)
Comment (bateman) [reply guidelines/_sources/02_report-version-and-configuration.tex:72]: Yes, indeed. I'm so sorry if the comment is misplaced.

**Review by sbaltes — CHANGES_REQUESTED** [review]

File: guidelines/_sources/02_report-version-and-configuration.tex (hunk 1 above)
Comment (bateman) [reply guidelines/_sources/02_report-version-and-configuration.tex:34]: I don't have a strong opinion, but yes, there's some overlapping, so I would merge and maybe treat fine-tuning as a 'special case' of recommendations?

File: guidelines/_sources/02_report-version-and-configuration.tex [hunk 6]
```diff
@@ -21,6 +21,28 @@ \subsubsection{Recommendations}
...
+\comment{This also leads me to another point. In my review of LLMs as annotators, I've found several examples of ensemble models. I think we might also want to add a section about ensemble models. For example:
+``When multiple models are used together (e.g., in ensemble or pipeline architectures), researchers \textbf{MUST} additionally:
+\begin{itemize}
+\item Report version and configuration details for each model separately;
+\item Describe the interaction between models;
+\item Explain the architecture for combining outputs (e.g., majority voting, weighted averaging, sequential processing);
+\item Document any logic that determines which model handles which inputs.
+\end{itemize}
+}
+
```
[the remark this answers is not in the document]
Comment (bateman) [reply guidelines/_sources/02_report-version-and-configuration.tex:45]: I see your point. Maybe we bring this to point to the whole team. To me, it seems that the two documents could be merged -- based on the current content. Unless we clarify and agree upon two clear-cut, differnt scopes

File: guidelines/_sources/02_report-version-and-configuration.tex (hunk 3 above)
Comment (bateman) [reply guidelines/_sources/02_report-version-and-configuration.tex:68]: Good point

File: guidelines/_sources/02_report-version-and-configuration.tex (hunk 4 above)
Comment (bateman) [reply guidelines/_sources/02_report-version-and-configuration.tex:72]: I'm moving it myself and marking this conversation as resolved.

File: guidelines/_sources/02_report-version-and-configuration.tex (hunk 1 above)
Comment (sbaltes) [reply guidelines/_sources/02_report-version-and-configuration.tex:34]: If we merge, we could have a dedicated paragraph on fine-tuning. Like: If the study involved fine-tuning a pretrained model, then..."

File: guidelines/_sources/02_report-version-and-configuration.tex (hunk 5 above)
Comment (bateman) [reply guidelines/_sources/02_report-version-and-configuration.tex:98]: I do like and support the idea of having checklist***s***. They make our guidelines easier and more 'actionable'

Comment (sbaltes) [reply guidelines/_sources/02_report-version-and-configuration.tex:98]: I've added a comment in the issue. In case you're fine continuing the discussion there, we can remove the comment here.

Comment (bateman) [reply guidelines/_sources/02_report-version-and-configuration.tex:98]: maybe comment out the comment? So that we remember what checklists we should provide?

Comment (sbaltes) [reply guidelines/_sources/02_report-version-and-configuration.tex:98]: Sounds good!

File: guidelines/_sources/02_report-version-and-configuration.tex (hunk 1 above)
Comment (profkalinowski) [reply guidelines/_sources/02_report-version-and-configuration.tex:34]: I agree with merging. There is already a dedicated paragraph on fine-tuning in the architecture section (we can, of course, combine them).

Similarly, if the LLM is fine-tuned, researchers \must describe the fine-tuning goal (\textit{e.g.}, domain adaptation, task specialization), procedure (\textit{e.g.}, full fine-tuning, parameter-efficient fine-tuning), and dataset (source, size, preprocessing, availability). They should include training details (\textit{e.g.}, compute resources, hyperparameters, loss function) and performance metrics (benchmarks, baseline comparison). If the data used for fine-tuning is not confidential, an anonymized snapshot of the data used for fine-tuning the model \should be made available.

Comment (sbaltes) [reply guidelines/_sources/02_report-version-and-configuration.tex:34]: @profkalinowski Once you've intergrated the remaining PR, I can rebase this branch on master: https://github.com/se-ubt/llm-guidelines-website/pull/49/files

**Comment by sbaltes** [comment]
This branch now contains the result of merging both guidelines. How do we proceed to consolidate the section?

**Comment by mircealungu** [comment]
Ehi @sbaltes - 'm sorry for being so slow to the conversation :( Conceptually, I think the fine-tuning belongs more to architecture and data. But I'll re-read the two sections again and will see whether that gives me new insight. At the same time, if you already decided to merge, that's fine too!

**Comment by sbaltes** [comment]
Sure, fine-tuning is rather related to architecture, but it is even more important in that case to clearly state the version of the underlying model. I don't see a good reason to keep the sections separated, but happy to be convinced otherwise :-)

**Comment by bateman** [comment]
My 2 cents is to keep them together.

**Comment by sbaltes** [comment]
Great, then we just need to decide how we merge the sections. The initial contributions for both sections were valuable, but I think we have to consolidate some of it to reduce redundancy and also make the section shorter.

File: guidelines/_sources/02_report-version-and-configuration.tex (hunk 1 above)
Comment (profkalinowski) [reply guidelines/_sources/02_report-version-and-configuration.tex]: Sorry for my late action, the previous PR is now integrated.

**Comment by sbaltes** [comment]
I've integrated the feedback into the main branch and will open a new PR once I start merging the sections
